code/modules/mob/living/brain/MMI.dm

```diff
@@ -127,25 +127,30 @@
 
 
 /obj/item/mmi/proc/transfer_identity(mob/living/L) //Same deal as the regular brain proc. Used for human-->robot people.
-	if(!brainmob)
-		brainmob = new(src)
-	brainmob.name = L.real_name
-	brainmob.real_name = L.real_name
-	if(L.has_dna())
-		var/mob/living/carbon/C = L
-		if(!brainmob.stored_dna)
-			brainmob.stored_dna = new /datum/dna/stored(brainmob)
-		C.dna.copy_dna(brainmob.stored_dna)
-	brainmob.container = src
-
 	if(ishuman(L))
 		var/mob/living/carbon/human/H = L
 		var/obj/item/organ/brain/newbrain = H.getorgan(/obj/item/organ/brain)
-		newbrain.forceMove(src)
-		brain = newbrain
-	else if(!brain)
+		if(newbrain)
+			. = TRUE
+			newbrain.Remove(H, TRUE) //this calls newbrain.transfer_identity()
+			newbrain.forceMove(src)
+			if(brain)
+				qdel(brain)
+			if(brainmob)
+				qdel(brainmob)
+			brain = newbrain
+
+	if(!brain)
 		brain = new(src)
-		brain.name = "[L.real_name]'s brain"
+
+	if(!brain.brainmob)
+		if(brainmob)
+			qdel(brainmob) //hopefully this isn't incredibly short sighted and ignorant and breaks everything
+		brain.transfer_identity(L)
```

Comment:
It's because brains don't create a brainmob when they're initialized, so we use transfer_identity() to make one

The qdel is a sanity check incase somehow the brain doesn't have a brainmob, but the MMI does, which shouldn't happen

Reply:
This does not account for roundstart (and latejoin) cyborgs. 

When any of these cyborgs has their MMI removed via any means it triggers a runtime and forcibly ejects the player to a ghost permanently.